### PR TITLE
Added more configurable filter options

### PR DESCRIPTION
**Pull request by rajan-nonstopio**

### CODE REVIEW

File: packages/syncfusion_flutter_datagrid/pubspec.lock [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Generated by pub
```
Comment (ProjectAJ14) [inline packages/syncfusion_flutter_datagrid/pubspec.lock]: Remove this

File: packages/syncfusion_flutter_datagrid/lib/src/datagrid_widget/widgets/cell_widget.dart [hunk 2]
```diff
@@ -697,7 +697,8 @@ class _FilterIcon extends StatelessWidget {
       // the issue by converting the global to local position of the current
       // overlay and used that new offset to display the show menu.
       final RenderBox renderBox =
-          Overlay.of(context).context.findRenderObject()! as RenderBox;
+          Overlay.of(context)?.context.findRenderObject() as RenderBox;
```
Comment (ProjectAJ14) [inline packages/syncfusion_flutter_datagrid/lib/src/datagrid_widget/widgets/cell_widget.dart:700]: Check what version of flutter was used here

Comment (ProjectAJ14) [reply packages/syncfusion_flutter_datagrid/lib/src/datagrid_widget/widgets/cell_widget.dart:700]: We may not need to change this

Comment (rajan-nonstopio) [reply packages/syncfusion_flutter_datagrid/lib/src/datagrid_widget/widgets/cell_widget.dart:700]: This throws issue if we dont add condition for this

**Pull request merged**